gen3-integration-tests/pages/indexing_page.py

```diff
@@ -0,0 +1,100 @@
+# Indexing Page
+import os
+import pytest
+
+from utils import logger
+from playwright.sync_api import expect, Page
+from utils.test_execution import screenshot
+from utils import TEST_DATA_PATH_OBJECT
+
+
+class IndexingPage(object):
+    def __init__(self):
+        # Endpoints
+        self.BASE_URL = f"{pytest.root_url_portal}/indexing"
+        # Locators
+        self.INDEXING_PAGE = "//div[@class='indexing-page']"  # Indexing page
+        self.UPLOAD_MANIFEST = (
+            "(//div[@class='action-panel'])[1]"  # UPLOAD_MANIFEST_BLOCK
+        )
+        self.DOWNLOAD_MANIFEST = (
+            "(//div[@class='action-panel'])[2]"  # DOWNLOAD_MANIFEST_BLOCK
+        )
+        self.POP_UP_BOX = "//div[@class='popup__box']"
+        # Locators - upload manifest
+        self.UPLOAD_MANIFEST_FORM = (
+            "//form[@class='index-flow-form']"  # MANIFEST_UPLOAD_FORM
+        )
+        self.FILE_UPLOAD = "//form[@class='index-flow-form']"
+        self.FILE_INPUT = "//input[@type='file']"
+        self.INDEX_FILE_BUTTON = "//button[contains(text(),'Index Files')]"
+        self.DOWNLOAD_LOGS_BUTTON = "//button[contains(text(), 'Download Logs')]"
+        self.FAILED_STATUS = "//div[@class='index-files-popup-text']/descendant::p[2]"
+        self.ERROR_IMAGE = "//*[name()='circle' and @id='Oval']"
+        # Locators - download manifest
+        self.DOWNLOAD_BUTTON = "//button[contains(text(),'Download')]"
+        self.SUCCESS_GREEN_MESSAGE = "//span[@class='index-files-green-label']"
+        self.DOWNLOAD_MANIFEST_BUTTON = (
+            "//button[contains(text(), 'Download Manifest')]"
+        )
+
+    def go_to(self, page: Page):
+        page.goto(self.BASE_URL)
+        screenshot(page, "IndexingPage")
+        page.wait_for_selector(self.INDEXING_PAGE, state="visible")
+        page.wait_for_selector(self.UPLOAD_MANIFEST, state="visible")
+        page.wait_for_selector(self.DOWNLOAD_MANIFEST, state="visible")
```

Comment:
we can identify which is the last to appear and use that as the READY_CUE to keep it consistent across

Reply:
I am sure how to define READY_CUE here? b/c we would need to have both upload section and download section to be visible. if not here in page.goto(), then we would have to check action panel for upload and download before we start to work on it. So I just added it here in page.goto function